src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c

```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+                }
+
+                if (phi_f2_2 < 0.0f) {
+                    phi_f2_2 = 0.0f;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Jsjutan/func_80A89860.s")
+                // temp_f14_2 = (spC8[i] * phi_f2_2) + ((this->unk_170 - (this->unk_170 * phi_f2_2)) - 200.0f);
+                temp_f14_2 = (spC8[i] * phi_f2_2);
+                temp_f14_2 += ((this->unk_170 - (this->unk_170 * phi_f2_2)) - 200.0f);
 
+                distance_2 = distance_2 - 1500.0f;
+                if (distance_2 < 0.0f) {
+                    distance_2 = 0.0f;
+                }
+
+                phi_f2_3 = 100.0f * distance_2 * 0.01f;
+                if (phi_f2_3 > 100.0f) {
+                    phi_f2_3 = 100.0f;
+                }
+
+                if (temp_f14_2 < phi_f28) {
+                    phi_f28 = temp_f14_2;
+                }
+
+                if (phi_f2_3 < spA8) {
+                    spA8 = phi_f2_3;
+                }
+            }
+        }
+
+        temp_f0_3 = Math_SinS(globalCtx->gameplayFrames * 4000 + j * 10000);
+
+        if (this->unk_174) {
+            temp_f14_2 = temp_f0_3 * spA8;
+
+            phi_v1_4 = phi_f28 + (temp_f14_2);
+            temp_a0_3 = (phi_s3->n.ob[1] - this->unk_168) * 50.0f;
+
+            if (phi_v1_4 < temp_a0_3) {
+                phi_v1_4 = temp_a0_3;
+            }
+
+            phi_s0->n.ob[1] = phi_v1_4;
+        } else {
+            temp_f14_2 = temp_f0_3 * spA8;
+
+            phi_s0->n.ob[1] = phi_f28 + (temp_f14_2);
+
+            phi_v1_4 = temp_f14_2 * 0.5f;
+            phi_s0->n.ob[0] = D_80A8EE10[j].x + phi_v1_4;
+            phi_s0->n.ob[2] = D_80A8EE10[j].z + phi_v1_4;
+
+            phi_v1_4 = temp_f14_2;
+            phi_s3->n.ob[0] = D_80A8EE10[j].x + phi_v1_4;
+            phi_s3->n.ob[2] = D_80A8EE10[j].z + phi_v1_4;
+        }
+
+        phi_s0++;
+        phi_s3++;
+    }
+
+    // address: ac4 ~ bb8
+    if (!this->unk_174) {
+        this->dyna.actor.velocity.y = 0.0f;
+        this->dyna.actor.world.pos.y = this->unk_168;
+
+        dayTime = gSaveContext.dayTime;
+        if (dayTime >= 0x8000) {
+            dayTime = 0xFFFF - dayTime;
+        }
+
+        this->shadowAlpha = (dayTime * 0.00275f) + 10.0f; // (1.0f / 364.0f) ?
+        this->unk_170 = 1000.0f;
+    } else {
+        Math_ApproachF(&this->dyna.actor.world.pos.y, this->unk_168 - 1000.0f, 1.0f, this->dyna.actor.velocity.y);
+        if (!gSaveContext.dayTime) {
+            if (1) {}
+            if (1) {}
+            if (1) {}
+            if (1) {}
+        }
+        Math_ApproachF(&this->dyna.actor.velocity.y, 5.0f, 1.0f, 0.5f);
+        Math_ApproachF(&this->shadowAlpha, 0.0f, 1.0f, 3.0f);
+        Math_ApproachF(&this->unk_170, -5000.0f, 1.0f, 100.0f);
+    }
+
+    // address: bbc
+    phi_s0_3 = phi_s0_2;
+
+    if (1) {}
+    if (1) {}
+
+    sp108.x = 0.0f;
+    sp108.y = 0.0f;
+    sp108.z = 120.0f;
+
+    for (j = 0; j < 0x90; j++) {
```

Comment:
All of these carpet-related numbers should be in decimal, as the carpet is 12 x 12.